src/plugins/discover/public/components/field_name/field_name.tsx

```diff
@@ -17,51 +17,32 @@
  * under the License.
  */
 import React from 'react';
-import classNames from 'classnames';
 import { EuiFlexGroup, EuiFlexItem, EuiToolTip } from '@elastic/eui';
 
 import { FieldIcon, FieldIconProps } from '../../../../kibana_react/public';
 import { shortenDottedString } from '../../helpers';
 import { getFieldTypeName } from './field_type_name';
 
-// property field is provided at discover's field chooser
 // properties fieldType and fieldName are provided in kbn_doc_view
 // this should be changed when both components are deangularized
 interface Props {
-  field?: {
-    type: string;
-    name: string;
-    rowCount?: number;
-    scripted?: boolean;
-  };
-  fieldName?: string;
-  fieldType?: string;
+  fieldName: string;
+  fieldType: string;
   useShortDots?: boolean;
   fieldIconProps?: Omit<FieldIconProps, 'type'>;
 }
 
-export function FieldName({ field, fieldName, fieldType, useShortDots, fieldIconProps }: Props) {
-  const type = field ? String(field.type) : String(fieldType);
+export function FieldName({ fieldName, fieldType, useShortDots, fieldIconProps }: Props) {
+  const type = String(fieldType);
```

Comment:
if the `fieldType` is string (and it's 100% string since it's passed through `DocViewTable` as String(fieldType)), so here is an extra conversion

Reply:
thx, removed that